src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+package parser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import command.Command;
+import command.DeadlineCommand;
+import command.DeleteAllCommand;
+import command.DeleteCommand;
+import command.DoneAllCommand;
+import command.DoneCommand;
+import command.EventCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.HelpCommand;
+import command.ListCommand;
+import command.ShowAfterCommand;
+import command.ShowBeforeCommand;
+import command.TodoCommand;
+import command.WrongCommand;
+import exception.DescriptionException;
+import exception.DukeDateTimeParserException;
+import exception.DukeKeywordException;
+import exception.NoIndexException;
+import task.DeadlineTask;
+import task.EventTask;
+import task.Task;
+import task.TodoTask;
+
+
+/**
+ * The Parser class are used to parse any user input and process
+ * them into executable command and data.
+ */
+public class Parser {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    /**
+     * An empty constructor to initialize Parser object.
+     */
+    public Parser() {}
+
+    /**
+     * This method parse a file path into parent directory and the file name.
+     *
+     * @param filepath String The path of a file
+     * @return String[] array of separated String.
+     */
+    public static String[] fileParser(String filepath) {
+        return filepath.split("/");
+    }
+
+
+    /**
+     * This method parse the content of a file line by line and
+     * turns into the correct Task object.
+     *
+     * @param line String Each line of strings in the file.
+     * @return Task Return the correct Task type from reading the file.
+     */
+    public static Task readFileParser(String line) {
+        String[] strings = line.split("\\|", 3);
+        String taskType = strings[0].trim();
+
+        switch(taskType) {
+        case "T" :
```

Comment:
I am not sure whether there should be a space between the case content and the colon here.